qa/tasks/salt_manager.py

```diff
@@ -78,13 +78,22 @@ def __cat_file_remote(self, remote, filename=None):
                 ).format(filename, remote.name))
 
     def __ping(self, ping_cmd, expected):
-        with safe_while(sleep=15, tries=50,
-                        action=ping_cmd) as proceed:
-            while proceed():
-                responded = len(re.findall('  True', self.master_remote.sh(ping_cmd)))
-                log.info("{} of {} minions responded".format(responded, expected))
-                if (expected == responded):
-                    return None
+        try:
+            def instances_of_str(search_str, output):
+                return len(re.findall(search_str, output))
+            with safe_while(sleep=15, tries=50,
+                            action=ping_cmd) as proceed:
+                while proceed():
+                    output = self.master_remote.sh(ping_cmd)
+                    no_master = instances_of_str('The salt master could not be contacted')
```

Comment:
You do not pass output to instances_of_str, that will make a exception?

Reply:
Good catch